FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -150,40 +156,46 @@ await foreach (var input in inputs.WithCancellation(cancellationToken))
             catch (TaskCanceledException taskCanceledEx)
             {
                 _diagnosticLogger.LogError("FhirToDataLake job is canceled.");
-                _logger.LogInformation(taskCanceledEx, "FhirToDataLake job {0} is canceled.", _jobInfo.Id);
+                _logger.LogInformation(taskCanceledEx, "FhirToDataLake orchestrator job {0} is canceled.", _jobInfo.Id);
+                _jobExecutionErrorProcessor.Process(taskCanceledEx, "FhirToDataLake orchestrator job is canceled.");
                 throw new RetriableJobException("Job is cancelled.", taskCanceledEx);
             }
             catch (OperationCanceledException operationCanceledEx)
             {
                 _diagnosticLogger.LogError("FhirToDataLake job is canceled.");
-                _logger.LogInformation(operationCanceledEx, "FhirToDataLake job {0} is canceled.", _jobInfo.Id);
+                _logger.LogInformation(operationCanceledEx, "FhirToDataLake orchestrator job {0} is canceled.", _jobInfo.Id);
+                _jobExecutionErrorProcessor.Process(operationCanceledEx, "FhirToDataLake orchestrator job is canceled.");
                 throw new RetriableJobException("Job is cancelled.", operationCanceledEx);
             }
             catch (SynapsePipelineExternalException synapsePipelineRetriableEx)
             {
                 // Customer exceptions.
                 _diagnosticLogger.LogError($"Error in FhirToDataLake job. Reason:{synapsePipelineRetriableEx.Message}");
                 _logger.LogInformation(synapsePipelineRetriableEx, "Error in orchestrator job {0}. Reason:{1}", _jobInfo.Id, synapsePipelineRetriableEx);
+                _jobExecutionErrorProcessor.Process(synapsePipelineRetriableEx, "Error in orchestrator job.");
```

Comment:
Shall we add details about the exception message in the message field?

Reply:
Will add details in processor.